app/assets/src/components/layout/Accordion.jsx

```diff
@@ -8,10 +8,11 @@ class Accordion extends React.Component {
     open: false
   };
 
-  componentDidMount() {
-    this.setState({
-      open: this.props.open
-    });
+  static getDerivedStateFromProps(newProps, prevState) {
+    return {
+      ...prevState,
+      open: newProps.open
```

Comment:
I think this is going to cause problems for the uncontrolled case. (where `<Accordion>` is keeping track of its own `open` state and you don't pass in `props.open`)

If you don't pass in `props.open`, then whenever any props change (including `props.children`), this function is going to overwrite `this.state.open` with `undefined`, even if it had been `true`. This will cause the accordion to unintentionally close.

I think you made a change in `d1eef6320` that broke this component and that the code before was correct. It looks like you were trying to set the `<Accordion>` to be open at the start instead of closed. In this case, you should add a separate `defaultOpen` prop, and do

```
constructor(props) {
  super(props);
  this.state = {
    open: this.props.defaultOpen || false,
  };
}
```
to initialize the internal `open` state to the value you want.

Let me know if this doesn't make sense.

Reply:
Very true, this is wrong. I'll fix it. Thanks for the explanation.

Reply:
I figured out a solution with the special `key` prop of react.